cmd/backup/main.go

```diff
@@ -283,28 +291,43 @@ func (s *script) registerHook(level hookLevel, action func(err error) error) {
 	s.hooks = append(s.hooks, hook{level, action})
 }
 
-// notifyFailure sends a notification about a failed backup run
-func (s *script) notifyFailure(err error) error {
-	body := fmt.Sprintf(
-		"Running docker-volume-backup failed with error: %s\n\nLog output of the failed run was:\n\n%s\n", err, s.output.String(),
-	)
-	title := fmt.Sprintf("Failure running docker-volume-backup at %s", s.start.Format(time.RFC3339))
+// notify sends a notification using the given title and body templates.
+// Automatically adds common params
+func (s *script) notify(titleTemplate string, bodyTemplate string, params map[string]string) error {
+	params["StartTime"] = s.start.Format(time.RFC3339)
```

Comment:
This is just discussing options: instead of using a `map[string]string` to hold the data that is passed to the templates, we could also define a struct type we then pass down. This could look for example like:

```
type NotificationData struct {
  StartTime string
  LogOuptut string
  Error error
  Config *config // this would make it really easy to access all config in the templates
}
```
That's probably a bit more verbose than using an ad-hoc map, but it might be easier to maintain as it implicitly documents what users can use in templates. Note to myself: this would also mean the definition of the `config` type becomes public API surface.

Reply:
I agree on creating a dedicated `struct` for this purpose.
Regarding passing the whole config, it's up to you if you want it to make public API. I'll wait for your decision :)

Reply:
I think if we start offering that feature, let's make it as useful as possible, and to me this also includes making the entire config avalilable. Considering the `ENV_VAR_NAMES` are public API anyways, i guess it's ok to keep the internal names locked as well. Let's do it.

Reply:
Perfect! One little question: I noticed that all the structs are lowercase in the project, but in your example `NotificationData` starts with an uppercase letter. Is this ok? What is the convention here?

Reply:
Docker for example also uses all uppercase in their templates i.e. `{{ .Container.Name }}`